lib/__mocks__/asyncBot.js

```diff
@@ -1,10 +1,9 @@
-// Allow tests to set return function as required
-const genAsyncBot = bot => ({
+const mockAsyncBot = {
   api: {
     users: {
-      info: bot.asyncBotReturn || jest.fn(),
+      info: jest.fn(),
     },
   },
-});
+};
 
-module.exports = genAsyncBot;
+module.exports = jest.fn(() => mockAsyncBot);
```

Comment:
Also really important here, the reason this actually works as a mock is because the implementation is _passing by reference_ not creating a new object every time.

This would *not* work:
```javascript
module.exports = jest.fn(() => ({
  api: {
    users: {
      info: bot.asyncBotReturn || jest.fn(),
      info: jest.fn(),
    },
  },
});
}));
```

Reply:
It's all the same code but when the genAsyncBot function is executed in the test, the mocks are applied to it.
The same func is then run in the implementation and a new instance is returned not having any of the mocks.

Pass by ref allows us to mock the behaviour and have the config persist on subsequent calls. Make sense?